SiameseDataset.py

```diff
@@ -0,0 +1,271 @@
+from __future__ import print_function
+import collections
+import h5py
+import numpy.random as random
+import numpy as np
+
+from CachedDataset2 import CachedDataset2
+from HDFDataset import FeatureSequenceStreamParser, SparseStreamParser, SegmentAlignmentStreamParser
+from Dataset import DatasetSeq
+
+
+
+class SiameseHDFDataset(CachedDataset2):
+  """
+  SiameseHDFDataset class allows to do sequence sampling for weakly-supervised training
+  It accepts data in the format of NextGenHDFDataset and performs sampling of sequence triplets before each epoch
+  Here we assume that each dataset from the input .hdf has a single label
+  Sampled triplets are splitted into different streams
+  In the config we can access streams by e.g. ["data:features_0", "data:features_1", "data:features_2"],
+  "data:features_0" stands for features of all sequences from the input data,
```

Comment:
So, multiple sequences? Or what is "all sequences"? How many sequences? And they are concatenated?
(All this should be clear from the documentation.)

Reply:
Also, why do you call it `features_0`? Why not `features_all` or sth like that? The name should better reflect what data the data stream contains.

Reply:
If concatenated, concatenated in time? So you end up with a single sequence of shape (sum_of_time, dimension)?

Reply:
If concatenated, how would you be able to split them later? Or you never need that?